src/main/scala/com/databricks/spark/redshift/RedshiftWriter.scala

```diff
@@ -311,29 +311,25 @@ private[redshift] class RedshiftWriter(
       // for a description of the manifest file format. The URLs in this manifest must be absolute
       // and complete.
 
-      // The saved filenames are going to be of the form part-r-XXXXX-UUID.avro. There isn't a way
-      // to figure out what this UUID is, besides querying S3. In principle, this file format could
-      // change in a way that breaks this library, but this seems unlikely and we can patch around
-      // it if that happens.
+      // The saved filenames depend on the spark-avro version. In spark-avro 1.0.0, the write
+      // path uses SparkContext.saveAsHadoopFile(), which produces filenames of the form
+      // part-XXXXX.avro. In spark-avro 2.0.0+, the partition filenames are of the form
+      // part-r-XXXXX-UUID.avro.
       val fs = FileSystem.get(URI.create(tempDir), sqlContext.sparkContext.hadoopConfiguration)
-      val writeJobUUID = fs.listStatus(new Path(tempDir))
```

Comment:
Actually, this shouldn't be a problem: if you have _millions_ of partitions then it almost certainly makes more sense to coalesce them prior to writing to Redshift. I can add this auto-coalescing in a followup.